### PR TITLE
Create authorization policies

### PR DESCRIPTION
**Pull request by NetanelBollag**
### Description

Create istio `AuthorizationPolicy` CRD for every intent. 
This PR also increments the Go version to 1.19 and Kubernetes controller-runtime version due to istio API dependencies. The controller-runtime update required also updating the controller option from `bool` to `*bool`, and that's the reason behind all the `RecoverPanic: lo.ToPtr(true)})` changes.

### Testing

- [x] This change adds test coverage for new/changed/fixed functionality

### CODE REVIEW

File: src/operator/controllers/external_traffic/endpoints_reconciler.go [hunk 1]
```diff
@@ -130,7 +132,8 @@ func (r *EndpointsReconciler) InitIngressReferencedServicesIndex(mgr ctrl.Manage
 		v1alpha2.IngressServiceNamesIndexField,
 		func(object client.Object) []string {
 			ingress := object.(*v1.Ingress)
-			return serviceNamesFromIngress(ingress).List()
+			services := serviceNamesFromIngress(ingress)
+			return sets.List(services)
```
Comment (NetanelBollag) [inline src/operator/controllers/external_traffic/endpoints_reconciler.go:136]: On Go 1.19 the sets.Strings package is deprecated, this change upgrades to the new generic Set.

File: src/operator/controllers/external_traffic/endpoints_reconciler.go [hunk 2]
```diff
@@ -47,7 +49,7 @@ func (r *EndpointsReconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&corev1.Endpoints{}).
-		WithOptions(controller.Options{RecoverPanic: true}).
+		WithOptions(controller.Options{RecoverPanic: lo.ToPtr(true)}).
 		Complete(r)
```
Comment (NetanelBollag) [inline src/operator/controllers/external_traffic/endpoints_reconciler.go:53]: All changes of `RecoverPanic` are a result of `sigs.k8s.io/controller-runtime` upgrade.

File: src/operator/controllers/external_traffic/ingress_reconciler.go [hunk 3]
```diff
@@ -67,8 +68,8 @@ func (r *IngressReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 	return r.reconcileIngressCreateOrUpdate(ctx, ingress)
 }
 
-func (r *IngressReconciler) getServicesReferencedByNetworkPoliciesCreatedForIngress(ctx context.Context, ingressName types.NamespacedName) (sets.String, error) {
-	services := sets.NewString()
+func (r *IngressReconciler) getServicesReferencedByNetworkPoliciesCreatedForIngress(ctx context.Context, ingressName types.NamespacedName) (sets.Set[string], error) {
+	services := sets.Set[string]{}
```
Comment (NetanelBollag) [inline src/operator/controllers/external_traffic/ingress_reconciler.go:73]: Same as above, replacing deprecated library

File: src/operator/controllers/kafkaacls/intentsAdmin.go [hunk 4]
```diff
@@ -76,7 +75,7 @@ func getTLSConfig(tlsSource otterizev1alpha2.TLSSource) (*tls.Config, error) {
 	}
 
 	pool := x509.NewCertPool()
-	rootCAPEM, err := ioutil.ReadFile(tlsSource.RootCAFile)
+	rootCAPEM, err := os.ReadFile(tlsSource.RootCAFile)
 	if err != nil {
```
Comment (NetanelBollag) [inline src/operator/controllers/kafkaacls/intentsAdmin.go:79]: The package `ioutil` is deprecated, replaced by os which according to documentation has the same implementation

**Review by omris94 — APPROVED** [review]

**Pull request merged**